### ISSUE TITLE
gollectd panics if insufficient data available in buffer

### ISSUE THREAD

**Issue by otoolep** [kimor79/gollectd#6]
https://github.com/kimor79/gollectd/blob/cf6dec97343244b5d8a5485463675d42f574aa2d/gollectd.go#L125

will panic if less than 2 bytes are available to be read. This occurs during a port scan of a collectd UDP input. See https://github.com/influxdb/influxdb/issues/3008 for details on how to reproduce this issue.

**Comment by otoolep** [comment]
Trace:

```
panic: runtime error: index out of range

goroutine 42 [running]:
github.com/kimor79/gollectd.Packets(0xc208138000, 0x1, 0x5ac, 0xc208151740, 0xc208107da0, 0x0, 0x0)
        /home/philip/repos/influxdb/src/github.com/kimor79/gollectd/gollectd.go:126 +0x2017
github.com/influxdb/influxdb/services/collectd.(*Service).handleMessage(0xc2080ae400, 0xc208138000, 0x1, 0x5ac)
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/services/collectd/service.go:187 +0x61
github.com/influxdb/influxdb/services/collectd.(*Service).serve(0xc2080ae400)
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/services/collectd/service.go:181 +0x268
created by github.com/influxdb/influxdb/services/collectd.(*Service).Open
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/services/collectd/service.go:110 +0xc28

goroutine 1 [chan receive (nil chan)]:
main.(*Main).Run(0xc2080a7f60, 0xc20800a000, 0x0, 0x0, 0x0, 0x0)
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/cmd/influxd/main.go:75 +0x450
main.main()
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/cmd/influxd/main.go:35 +0x1d0

goroutine 6 [syscall]:
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2232 +0x1
```

**Comment by kimor79** [comment]
Fixed with #7
